src/python/pants/build_graph/mutable_build_graph.py

```diff
@@ -74,14 +76,21 @@ def inject_address_closure(self, address):
 
       if not self.contains_address(target_address):
         target = self._target_addressable_to_target(target_address, target_addressable)
+        target.apply_injectables(self)
         self.inject_target(target, dependencies=dep_addresses)
       else:
         for dep_address in dep_addresses:
           if dep_address not in self.dependencies_of(target_address):
             self.inject_dependency(target_address, dep_address)
         target = self.get_target(target_address)
 
-      for traversable_spec in target.traversable_dependency_specs:
+      traversables = [target.compute_dependency_specs(payload=target.payload)]
+      # Only poke `traversable_dependency_specs` if a concrete implementation is defined
+      # in order to avoid spurious deprecation warnings.
```

Comment:
the warning will be seen with a`super()` call to the parents `traversable_dependency_specs`.

putting it there has 2 benefits that I can see:

1) it's closer to the code itself - e.g. just above the method definition where the `:API public:` exists, which is easier for casual readers to spot.
2) we don't have to duplicate a `deprecated_conditional` in the 3 different places where t_d_s is poked.

tho it's possible that external consumers might forget to call `super()`, I don't think it's likely? but open to that approach if you think it's more reasonable.